packages/react-components/src/components/emoji-variations-list.tsx

```diff
@@ -0,0 +1,200 @@
+`
+
+const Emoji = styled(Gif)`
+    display: inline-block;
+    flex-shrink: 0;
+    list-style: none;
+    /* make sure gifs are fully visible with a scrollbar */
+    margin-bottom: 1px;
+    position: relative;
+
+    .${Gif.imgClassName} {
+        position: absolute;
+        top: 0;
+        left: 0;
+    }
+`
+
+export function EmojiVariationsList(props: EmojiVariationsListProps) {
+    const {
+        backgroundColor = DEFAULT_BG_COLOR,
+        className,
+        dividerColor = DEFAULT_DIVIDER_COLOR,
+        fetchVariations,
+        gif,
+        gifHeight,
+        GifProps = {},
+        gifWidth: gifWidthProp,
+        gutter = 6,
+        hideAttribution,
+        loader: LoaderComponent,
+        noLink,
+        onGifClick,
+        onGifKeyPress,
+        onGifRightClick,
+        onGifSeen,
+        onGifVisible,
+        onVariationsFetched,
+        overlay,
+        tabIndex = 0,
+        user,
+        ...other
+    } = props
+
+    const variationCount = gif.variation_count ?? 0
+    const gifWidth = gifWidthProp || getGifWidth(gif, gifHeight)
+    const variationsContainerWidth = variationCount * gifWidth + Math.max(0, variationCount - 1) * gutter
+
+    const fetchVariationsRef = useLatest(fetchVariations)
+    const onVariationsFetchedRef = useLatest(onVariationsFetched)
+    const [variationsState, syncVariations] = useAsyncFn(
+        async () => {
+            const result = await fetchVariationsRef.current(gif.id)
+            const variations = (result?.data ?? []) as IGif[]
+            if (onVariationsFetchedRef.current) {
+                onVariationsFetchedRef.current(variations)
+            }
+            return variations
+        },
+        [gif],
+        { loading: true, value: [] }
+    )
+
+    useEffect(() => {
+        // noinspection JSIgnoredPromiseFromCall
+        syncVariations()
+    }, [syncVariations])
+
+    const emojiCommonProps = {
+        backgroundColor: 'transparent',
+        height: gifHeight,
+        hideAttribution,
+        noLink,
+        onGifClick,
+        onGifKeyPress,
+        onGifRightClick,
+        onGifSeen,
+        onGifVisible,
+        overlay,
+        tabIndex,
+        user,
+        width: gifWidth,
+        ...GifProps,
+    }
+
+    const getVariationsContent = () => {
+        if (variationsState.loading) {
+            return LoaderComponent ? <LoaderComponent className={EmojiVariationsList.loaderClassName} /> : null
+        }
+        if (variationsState.error) {
+            return <FetchError onClick={syncVariations} />
+        }
+
+        return (
+            <>
+                {(variationsState.value ?? []).map((gif) => (
+                    <Emoji {...emojiCommonProps} key={gif.id} gif={gif} />
+                ))}
+            </>
+        )
+    }
+
+    return (
+        <Root
+            backgroundColor={backgroundColor}
+            className={[EmojiVariationsList.className, className].join(' ')}
+            {...other}
```

Comment:
@pshoniuk this seems a little loose, is it is supposed to be a way to add `style`? do we need it or is the `className` sufficient for styling?

Reply:
@giannif No, it is not explicitly designed for style. It is added to support standard HTML element props. It seems native because EmojiVariationsList is a visual element and is very convenient for the end user because, for example, it can add id, aria-accessibility attributes, event listeners, etc., without unnecessary wrappers or code modification. On the other hand, it doesn't really affect the support of the library from our side.

Reply:
@pshoniuk okay, that makes sense, I've never had to do that with a visual component but you're right it doesn't have any drawbacks